plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveInputInfo.java

```diff
@@ -17,20 +17,26 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 import java.util.List;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
 
 public class HiveInputInfo
 {
     private final List<String> partitionIds;
+    private final Optional<String> fileFormat;
     // Code that serialize HiveInputInfo into log would often need the ability to limit the length of log entries.
     // This boolean field allows such code to mark the log entry as length limited.
     private final boolean truncated;
 
     @JsonCreator
     public HiveInputInfo(
             @JsonProperty("partitionIds") List<String> partitionIds,
+            @JsonProperty("fileFormat") Optional<String> fileFormat,
```

Comment:
Partitioned table may have multiple file formats (each partition can be different)

Reply:
Right, however I'm worried that that information may not be cheap to obtain. For our use case we're okay with just knowing about the current storage format configured for the table.
Would it be okay if we renamed `fileFormat` to `tableFileFormat` to be more explicit avoid the effort getting this info for all partitions ?

Reply:
> Would it be okay if we renamed `fileFormat` to `tableFileFormat` to be more explicit avoid the effort getting this info for all partitions ?

Seems like an OK workaround. Add code comment that's it's only as much useful for partitioned tables.

or call it `tableDefaultFileFormat`

Reply:
@martint we want to be able to do an offline analysis about the relative occurrence of each file format for reads. We're recording the default table format instead of extracting the storage format of every partition to avoid adding overheads for collecting this info and because knowing the table level format is good enough approximation for telemetry use cases.